src/Illuminate/Collections/Traits/EnumeratesValues.php

```diff
@@ -555,14 +555,23 @@ public function unlessNotEmpty(callable $callback, callable $default = null)
     /**
      * Filter items by the given key value pair.
      *
-     * @param  callable|string  $key
+     * @param  callable|string|array  $key
      * @param  mixed  $operator
      * @param  mixed  $value
      * @return static
      */
     public function where($key, $operator = null, $value = null)
     {
-        return $this->filter($this->operatorForWhere(...func_get_args()));
+        if (is_iterable($key)) {
+            $filteredCollection = $this;
+            foreach ($key as $searchKey => $searchValue) {
+                $filteredCollection = $filteredCollection->filter($filteredCollection->operatorForWhere($searchKey, null, $searchValue));
+            }
+
+            return new static($filteredCollection);
+        } else {
```

Comment:
There's no need for an else statement here as the previous if statement will always cause a return. So we can avoid nesting the last return inside an else statement here.

Reply:
Updated. Thanks for pointing out.